registration-processor/registration-processor-info-storage-service/src/test/java/io/mosip/registration/processor/packet/storage/utils/ABISHandlerUtilTest.java

```diff
@@ -1,164 +1,170 @@
+import io.mosip.registration.processor.packet.manager.idreposervice.IdRepoService;
+import io.mosip.registration.processor.packet.storage.dao.PacketInfoDao;
+import io.mosip.registration.processor.packet.storage.dto.ApplicantInfoDto;
+import io.mosip.registration.processor.packet.storage.mapper.PacketInfoMapper;
+import io.mosip.registration.processor.status.code.RegistrationStatusCode;
+import io.mosip.registration.processor.status.entity.RegistrationStatusEntity;
+
+/**
+ * The Class PacketInfoManagerImplTest.
+ */
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ Utilities.class, PacketInfoMapper.class })
+@PowerMockIgnore({ "javax.management.*", "javax.net.ssl.*","com.sun.org.apache.xerces.*", "javax.xml.*", "org.xml.*", "javax.management.*" })
+public class ABISHandlerUtilTest {
+
+	private static final String registrationId = "10002100820001420210108085956";
+	private static final String registrationType = "NEW";
+	private static final String latestTransactionId = "123-456-789";
+	List<String> matchedRids = new ArrayList<>();
+
+
+	@InjectMocks
+	private ABISHandlerUtil abisHandlerUtil;
+
+	@Mock
+	private Utilities utilities;
+
+	@Mock
+	private PacketInfoManager<Identity, ApplicantInfoDto> packetInfoManager;
+
+	@Mock
+	private PacketInfoDao packetInfoDao;
+
+	@Mock
+	private IdRepoService idRepoService;
+	
+
+
+	@Before
+	public void setup() throws Exception {
+		MockitoAnnotations.initMocks(this);
+		when(utilities.getLatestTransactionId(registrationId)).thenReturn(latestTransactionId);
+
+		List<String> regBioRefIds = new ArrayList<>();
+		regBioRefIds.add("cf1c941a-142c-44f1-9543-4606b4a7884e");
+
+		when(packetInfoDao.getAbisRefMatchedRefIdByRid(registrationId)).thenReturn(regBioRefIds);
+		when(utilities.getGetRegProcessorDemographicIdentity()).thenReturn(new String());
+		List<RegistrationStatusEntity> registrationStatusEntityList = new ArrayList<>();
+
+		RegistrationStatusEntity registrationEntity1 = new RegistrationStatusEntity();
+		registrationEntity1.setId("10002100820001420210108085103");
+		registrationEntity1.setStatusCode(RegistrationStatusCode.PROCESSED.toString());
+		registrationStatusEntityList.add(registrationEntity1);
+		RegistrationStatusEntity registrationEntity2 = new RegistrationStatusEntity();
+		registrationEntity2.setId("10002100820001420210108085100");
+		registrationEntity2.setStatusCode(RegistrationStatusCode.PROCESSING.toString());
+		registrationStatusEntityList.add(registrationEntity2);
+		RegistrationStatusEntity registrationEntity3 = new RegistrationStatusEntity();
+		registrationEntity3.setId("10002100820001420210108085102");
+		registrationEntity3.setStatusCode(RegistrationStatusCode.PROCESSED.toString());
+		registrationStatusEntityList.add(registrationEntity3);
+		matchedRids.add("10002100820001420210108085100");
+		matchedRids.add("10002100820001420210108085103");
+		matchedRids.add("10002100820001420210108085101");// REJECTED

```

Comment:
I dont see this 10002100820001420210108085101 being added to the registration set?

Reply:
Why I have not added is when(packetInfoDao.getWithoutStatusCode(matchedRids, RegistrationStatusCode.REJECTED.toString()))
				.thenReturn(registrationStatusEntityList); this should not return REJECTED records but it will be in matchedRids